### PR TITLE
Add support for querying encryption status from the AzureDiskEncryptionForLinux extension

### PR DESCRIPTION
**Pull request by krkhan**
Internal CR at CodeFlow:

codeflow:open?server=http://codeflow/Services/DiscoveryService.svc&review=kakhan-39e101815e2546efaaf9136c37643be7&alert=true

### CODE REVIEW

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 1]
```diff
@@ -45,6 +49,141 @@ public class GetAzureDiskEncryptionStatusCommand : VirtualMachineExtensionBaseCm
         [ValidateNotNullOrEmpty]
         public string VMName { get; set; }
 
+        [Alias("ExtensionName")]
+        [Parameter(
+            Mandatory = false,
+            Position = 2,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The extension name. If this parameter is not specified, default values used are AzureDiskEncryption for windows VMs and AzureDiskEncryptionForLinux for Linux VMs")]
```
Comment (ejarvi) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: capitalize "windows" -> "Windows"

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs [hunk 2]
```diff
@@ -209,17 +348,53 @@ public override void ExecuteCmdlet()
...
+                            this.Name,
+                            parameters).GetAwaiter().GetResult();
+
+                        Dictionary<string, string> encryptionStatusParsed = null;
+                        try
+                        {
+                            string encryptionStatusJson = GetExtensionStatusMessage(osType, returnSubstatusMessage: true);
+                            encryptionStatusParsed = JsonConvert.DeserializeObject<Dictionary<string, string>>(encryptionStatusJson);
+                        }
+                        catch(KeyNotFoundException)
```
Comment (ejarvi) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: Is this catch too narrow?  In the case of invalid input, it seems there might be other exceptions that could be thrown by this block that we might want to handle in the same way.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs (hunk 1 above)
Comment (krkhan) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: Fixed with d4d5774d

File: src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs (hunk 2 above)
Comment (krkhan) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/GetAzureDiskEncryptionStatus.cs]: This is the exception that we throw ourselves in GetExtensionStatusMessage:

```
                if (returnSubstatusMessage)
                {
                    if(context.SubStatuses != null)
                    {
                        return context.SubStatuses[0].Message;
                    }
                    else
                    {
                        throw new KeyNotFoundException(string.Format(CultureInfo.CurrentUICulture, "Invalid extension substatus"));
                    }
                }
```

For other exceptions, we probably want to fail like we would've before this substatus change.

**Comment by markcowl** [comment]
@krkhan please respond to review comments.

**Comment by krkhan** [comment]
@markcowl Addressed comments on both GH and CF. Added you to the CF review as well.

**Comment by markcowl** [comment]
on demand run here:http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1125/

LGTM once this passes

**Comment by krkhan** [comment]
@markcowl That error doesn't look related to disk encryption.

> d:\workspace\powershell-demand\src\ResourceManager\Sql\Commands.Sql.Test\ScenarioTests\SqlTestsBase.cs(76): error : Microsoft.Azure.Commands.Sql.Test.ScenarioTests.ThreatDetectionTests.ThreatDetectionGetDefualtPolicy [FAIL] [D:\workspace\powershell-demand\build.proj]

**Comment by markcowl** [comment]
@krkhan It isn't, you need to rebase this change on the latest in the dev branch, as there is a fix for this issue there.on the

**Comment by cormacpayne** [comment]
@krkhan Hey Kamran, the test that is failing in your on-demand build has been fixed and merged into dev, so if you pull and rebase with dev, you should be fine.

You can check out how to rebase in our [cleaning up commits guide](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md)

**Comment by markcowl** [comment]
@krkhan, we can't merge until you do this

**Comment by krkhan** [comment]
There's a minor CR change that's needed. I'll do that today as well as merge upstream dev so that the fix is pulled in.

**Comment by ejarvi** [comment]
@markcowl finished reviewing, looks good

**Comment by krkhan** [comment]
@markcowl The ondemand run you started succeeded: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1138/

**Pull request merged**